core/test/movegen_test.c

```diff
@@ -366,6 +367,108 @@ void macondo_tests(SuperConfig *superconfig) {
   destroy_game(game_two);
 }
 
+// print assertions to paste into leave_lookup_test
+void print_leave_lookup_test(SuperConfig *superconfig) {
+  Config *config = get_csw_config(superconfig);
+  Game *game = create_game(config);
+
+  char cgp[300] =
+      "ZONULE1B2APAID/1KY2RHANJA4/GAM4R2HUI2/7G6D/6FECIT3O/"
+      "6AE1TOWIES/6I7E/1EnGUARD6D/NAOI2W8/6AT7/5PYE7/5L1L7/"
+      "2COVE1L7/5X1E7/7N7 MOOORRT/BFQRTTV 340/419 0 lex CSW21;";
+  load_cgp(game, cgp);
+
+  double *leaves = game->gen->leave_map->leave_values;
+  // Initialize data to notice which elements are not set.
+  for (int i = 0; i < (1 << RACK_SIZE); i++) {
+    leaves[i] = DBL_MAX;
+  }
+  generate_leaves_for_game(game, true);
+  const char rack[8] = "MOOORTT";
+  for (int i = 0; i < (1 << RACK_SIZE); ++i) {
+    const double value = leaves[i];
+    if (value == DBL_MAX) {
+      continue;
+    }
+    char leave_string[8];
+    int k = 0;
+    for (int j = 0; j < RACK_SIZE; ++j) {
+      if (i & (1 << j)) {
+        leave_string[k++] = rack[j];
+      }
+      leave_string[k] = '\0';
+    }
+    printf("assert(within_epsilon(leaves[%3d], %+14.6f));", i, value);
+    printf(" // %s\n", leave_string);
+  }
+  destroy_game(game);
+}
+
+void leave_lookup_test(SuperConfig *superconfig) {
+  Config *config = get_csw_config(superconfig);
+  Game *game = create_game(config);
+
+  char cgp[300] =
+      "ZONULE1B2APAID/1KY2RHANJA4/GAM4R2HUI2/7G6D/6FECIT3O/"
+      "6AE1TOWIES/6I7E/1EnGUARD6D/NAOI2W8/6AT7/5PYE7/5L1L7/"
+      "2COVE1L7/5X1E7/7N7 MOOORRT/BFQRTTV 340/419 0 lex CSW21;";
+  load_cgp(game, cgp);
+
+  for (int add_exchanges = false; add_exchanges <= true; add_exchanges++) {
```

Comment:
IIUC, this relies on the fact that true is implemented as 1 in stdbool, but it could in theory be any nonzero value, so I think it's preferred just keep the for loop numerical and and use i == 0 as the boolean expression to pass to generate_leaves_for_game.

Reply:
done